usr/src/lib/libzfs/common/libzfs_sendrecv.c

```diff
@@ -2797,7 +2797,7 @@ zfs_receive_package(libzfs_handle_t *hdl, int fd, const char *destname,
 				*strchr(tofs, '@') = '\0';
 			}
 
-			if (recursive && !flags->dryrun && !flags->nomount) {
+			if (!flags->dryrun && !flags->nomount) {
```

Comment:
OK, I'm an idiot, diff reading comprehension problem.  Your change is great, sorry for the noise.